app/models/merchant.rb

```diff
@@ -32,6 +30,16 @@ def self.search_all(params)
     end
   end
 
+  def self.revenue(params)
+    revenue = Merchant.joins(invoices: [:transactions, :invoice_items]).
```

Comment:
I think it might be more conventional to put the periods at the beginning of the line instead of the end.